test/index.js

```diff
@@ -545,6 +545,109 @@ describe('def', function() {
                    'as its first argument; received ["abc"]'));
   });
 
+  it('supports the "ValidNumber" type', function() {
+    var def = $.create($.env.concat([$.ValidNumber]));
+
+    // add :: ValidNumber -> ValidNumber
```

Comment:
You mean `inc` rather than `add`. Also, it looks nice if the function name in the signature is aligned with the function name in the definition:

```javascript
//  inc :: ValidNumber -> ValidNumber
var inc = ...;
```